packages/core/types/src/analytics/mutations.ts

```diff
@@ -0,0 +1,45 @@
+export interface TrackAnalyticsEventDTO {
+  /**
+   * The event name
+   */
+  event: string
+  /**
+   * The actor of the event, if there is any
+   */
+  actor?: {
+    id: string
+  }
+  /**
+   * The group that the event is for, such as an organization or team.
+   * The "type" defines the name of the group (eg. "organization"), and the "id" is the id of the group.
+   */
+  group?: {
+    type?: string
+    id?: string
+  }
```

Comment:
nit/suggestion: This feels a little bit Posthog-specific; would it make more sense to add a `context` object, that can carry information specific to providers?

Reply:
There are multiple providers that support the same construct (eg. https://amplitude.com/docs/apis/analytics/group-identify) so I figured I should make it a first-class citizen. A lot of their APIs are the way they are due to a necessity, and I think it's better to have better/strongly typed support for it, WDYT?

Reply:
Yeah, I see. Fine to keep this as is.